src/platform_impl/android/mod.rs

```diff
@@ -169,6 +185,18 @@ impl<T: 'static> EventLoop<T> {
                             );
                         }
                     }
+                    Event::Destroy => {
+                        let event = event::Event::WindowEvent {
+                            window_id: window::WindowId(WindowId),
+                            event: event::WindowEvent::CloseRequested,
+                        };
+                        call_event_handler!(
+                            event_handler,
+                            self.window_target(),
+                            control_flow,
+                            event
+                        );
+                    }
```

Comment:
It seems wrong to send `CloseRequested` when the application is destroyed; maybe it should be the `Destroyed` event instead?

Reply:
If I get everthing correctly, there is no direct destroy in android. There is just a request to destroy the app. It's send to background, to make reopening faster. If there are not enough resources available, the app gets destroyed, but those events are handled by OS.

I will check this again!

Reply:
I checked this. I'm open to change this to WindowEvent::Destroyed. I think it's some sort of interpretation:
Is it asking for destroying or is it force destroying. 
When android receives this event, render context is still alive and can be saved or released gracefully, therefore i thought it as "the window is requested to be closed". You can also see this as forced shutdown.

If this is the only problem, i will change it.

Reply:
`Event::Destroy` happens when the entire activity is destroyed (https://developer.android.com/guide/components/activities/activity-lifecycle#ondestroy), theoretically the window should have been destroyed prior, with `Event::WindowDestroyed`?

This is dispatched as a `winit` `Suspended` event though. Android''s Suspend/Pause and Resume events represent something different than winit's `Suspended` / `Resumed` (specific for Android backends iirc).

Reply:
It would be best, if we can strictly divide Event and WindowEvent. The questio is, if winit fires all those events correctly, also in android environment. Again, we must not lose any important events for graphics handling.

Reply:
> Again, we must not lose any important events for graphics handling.

As said it'd be great if this event is handled properly in `winit`, but we must not diverge from `winit` patterns too far. It is already a stretch to have `Suspended` and `Resumed` (whether their naming is useful/accurate is still up for debate) as this makes `winit` applications for other platforms not immediately runnable on Android without refactoring swapchain-lifetime logic to adhere to these events.

(Tangent: theoretically it'd be nice if every platform could emit these events for swapchain creation and destruction, making the whole thing more generic. Even better: we provide the `RawWindowHandle` only in the `Resumed` [or otherwise named] (`Window`)`Event` so that crates are forced to handle this event instead of calling `fn raw_window_handle` at will.)

> I think it's some sort of interpretation:
> Is it asking for destroying or is it force destroying.

Regarding this prior question: Whenever this function/event is called, you are _supposed to_ release all resources related to the window (again: the swapchain) and return from this function (`ndk_glue` does that after the lock from `fn native_window()` is released, presuming the caller read the documentation and is already holding on to this lock while they have resources created on the window). In that sense you _could_ keep using and rendering to the swapchain without ever returning from `on_window_destroyed`, but I don't know what Android will do in this case (it's already out of focus and most likely also out of view, so it'll probably result in an ANR, and no new events will arrive).